doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+know that they always produce the output that they say that they do. Here is an
+`example docstring
+<https://github.com/sympy/sympy/blob/b176f6a1d9890b42dc361857c887992315e3d5ad/sympy/functions/elementary/complexes.py#L22-L47>`_.
+Most docstrings are also automatically included in the Sphinx documentation
+above, so that they appear on the SymPy Documentation website. Here is that
+`same docstring
+<https://docs.sympy.org/latest/modules/functions/elementary.html#sympy.functions.elementary.complexes.im>`_
+on the SymPy website. The docstrings are formatted in a specific way so that
+Sphinx can render them correctly for the docs website. The SymPy sources all
+contain sparse technical documentation in the form of source code comments,
+although this does not generally constitute anything substantial and is not
+displayed on the documentation website.
+
+**SymPy Wiki** https://github.com/sympy/sympy/wiki
+
+The SymPy Wiki can be edited by anyone without review. It contains various
+types of documentation, including:
+
+* High-level developer documentation (for example: https://github.com/sympy/sympy/wiki/Args-Invariant)
+* Guides for new contributors (for example: https://github.com/sympy/sympy/wiki/Introduction-to-contributing)
+* Development policies (for example: https://github.com/sympy/sympy/wiki/Python-version-support-policy)
+* Release notes (for example: https://github.com/sympy/sympy/wiki/Release-Notes-for-1.5)
+* Various pages that different contributors have added
+
+Narrative Documentation Guidelines
+==================================
+
+Extensive documentation, or documentation that is not centered around an API
+reference, should be written as a narrative document in the Sphinx docs (located
+in the `doc/src directory
+<https://github.com/sympy/sympy/tree/master/doc/src>`_). The narrative documents
+do not reside in the Python source files, but as standalone restructured files
+in the doc/src directory. SymPy’s narrative documentation is defined as the
+collective documents, tutorials, and guides that teach users how to use SymPy.
+
+Steps to Writing Documentation
+------------------------------
+
+The first step to contributing to the code base is creating your development
+environment. Please find instructions on how to create your development
+environment in our `Development Workflow – Create Your Environment
+<https://github.com/sympy/sympy/wiki/Development-workflow#create-your-environment>`_
+guide.
+
+Once you have created your development environment, follow these steps:
+
+1. Installation
+---------------
+
+**Debian/Ubuntu**
+
+For Debian/Ubuntu, install the prerequisites::
+
+   apt-get install python-sphinx texlive-latex-recommended dvipng librsvg2-bin
+   imagemagick docbook2x graphviz python -m pip install sphinx-math-dollar
+
+And do::
+
+   make html
+
+If you get mpmath error, install python-mpmath package::
+
+   apt-get install python-mpmath
```

Comment:
Maybe installing `mpmath` should be suggested before `make html` (same goes for matplotlib below). It is now straight-forward to get these with `pip` on most systems.

I personally set this up using `cd sympy; pip install -e .` which installs the local sympy and any dependencies (mpmath). I do this using virtual environments though which is probably overkill for the description here...

Reply:
IMO we shouldn't recommend pip install -e. Unless you do use some kind of virtual environment, this will effectively install the development sympy globally, which could break people's installations. It's better to just tell people to run things from within the sympy directory. Since SymPy is pure Python, there's no need to do any building or installation to use it from the dev directory.